instana/instrumentation/sqlalchemy.py

```diff
@@ -47,23 +50,43 @@ def receive_after_cursor_execute(**kw):
             if scope is not None:
                 scope.close()
 
-    @event.listens_for(Engine, 'dbapi_error', named=True)
-    def receive_dbapi_error(**kw):
-        context = kw['context']
 
-        if context is not None and hasattr(context, '_stan_scope'):
-            scope = context._stan_scope
-            if scope is not None:
-                scope.span.mark_as_errored()
+    error_event = "handle_error"
+    # Handle dbapi_error event; deprecated since version 0.9
+    if sqlalchemy.__version__[0] == "0":
+        error_event = "dbapi_error"
 
-                if 'exception' in kw:
-                    e = kw['exception']
-                    scope.span.set_tag('sqlalchemy.err', str(e))
-                else:
-                    scope.span.set_tag('sqlalchemy.err', "No dbapi error specified.")
-                scope.close()
 
+    def _set_error_tags(scope, exception):
+        scope.span.mark_as_errored()
```

Comment:
I know this goes a bit outside of current PR, since this was how it's been implemented originally, but usually we call `InstanaSpan.log_exception()` to add exceptions to the span. You might want to add a special handling for SQLAlchemy operation there though.